_projects/authenticated_3rd_party_invites.md

```diff
@@ -0,0 +1,104 @@
+---
+name: 3rd Party Authorised Room Membership
+desc: Invite and kick users in a room based on some external authorisation. 
+requirements:
+ - Understanding of the matrix Client-Server API
+ - Familiarity with Javascript & React
+ - Familiarity with a programming language with a good matrix SDK (for the
+   server side component).
+difficulty: medium
+issues:
+mentors:
+ - Cadair
+ - Half-Shot
+initiatives:
+ - GSoC
+tags:
+---
+
+#### Description
+
+This project aims to fill a gap which currently exists in the matrix
+ecosystem, which is the ability to delegate membership of a room to a third
+party service.
+
+Two examples of this workflow include:
+
+1. A (Patreon) supporters room: if you support a project at a certain level you
```

Comment:
One thing to keep in mind in the Patreon case: Patreon doesn't give people's MxID, but their email address. I don't think it's possible to add people to the Patreon supporters room directly without emailing people a link to a form so they give their MxID

Reply:
The example workflow is that you click a link in the Patreon, login to your matrix account and then the bot has a verified link between your Patreon account and your matrix id. Meaning the bot can auth you for invites and check up in the future on your Patreon status.

Reply:
I added a sentence about this to hopefully make it clearer.